internal/component/memberlease/memberlease.go

```diff
@@ -0,0 +1,157 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package memberlease
+
+import (
+	"fmt"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	"github.com/gardener/etcd-druid/internal/component"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/hashicorp/go-multierror"
+	coordinationv1 "k8s.io/api/coordination/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrListMemberLease indicates an error in listing the member lease resources.
+	ErrListMemberLease druidv1alpha1.ErrorCode = "ERR_LIST_MEMBER_LEASE"
+	// ErrSyncMemberLease indicates an error in syncing the member lease resources.
+	ErrSyncMemberLease druidv1alpha1.ErrorCode = "ERR_SYNC_MEMBER_LEASE"
+	// ErrDeleteMemberLease indicates an error in deleting the member lease resources.
+	ErrDeleteMemberLease druidv1alpha1.ErrorCode = "ERR_DELETE_MEMBER_LEASE"
+)
+
+type _resource struct {
+	client client.Client
+}
+
+// New returns a new member lease component operator.
+func New(client client.Client) component.Operator {
+	return &_resource{
+		client: client,
+	}
+}
+
+// GetExistingResourceNames returns the names of the existing member leases for the given Etcd.
+func (r _resource) GetExistingResourceNames(ctx component.OperatorContext, etcdObjMeta metav1.ObjectMeta) ([]string, error) {
+	resourceNames := make([]string, 0, 1)
+
+	objMetaList := &metav1.PartialObjectMetadataList{}
+	objMetaList.SetGroupVersionKind(coordinationv1.SchemeGroupVersion.WithKind("Lease"))
+	if err := r.client.List(ctx,
+		objMetaList,
+		client.InNamespace(etcdObjMeta.Namespace),
+		client.MatchingLabels(getSelectorLabelsForAllMemberLeases(etcdObjMeta)),
+	); err != nil {
```

Comment:
Because `memberlease` component can consist of multiple member lease resources, hence it's a `List()` call, not a `Get()` call, unlike the other components. A Get() call returns a `NotFound` error if the resource does not exist, and needs to be explicitly handled, rather than returning the error as is. Whereas, a `List()` call is for a `*List` resource, like a `LeaseList` (technically, `PartialObjectMetadataList` here, since we only want metadata of the leases). If no leases are found, then the `Items` slice in the `PartialObjectMetadataList` will be empty, and error is `nil`. So we can safely return the items as is, without any error, as long as no other error was encountered by the `List()` call.

Reply:
Ah, you're right. My bad
Resolving this